lib/rspec/matchers/built_in/include.rb

```diff
@@ -15,12 +15,14 @@ def initialize(*expecteds)
         # @api private
         # @return [Boolean]
         def matches?(actual)
+          actual = actual.to_hash if actual.respond_to?(:to_hash)
```

Comment:
Can you use `Hash.try_convert` instead?  Seems purpose-built for when you want to try to convert an object to a hash...

Reply:
I don't follow. `Hash.try_convert` returns `nil` on failure, which we don't want here, as `include` does accept things other than hashes.

Unless you mean to do the following?

```ruby
actual = Hash.try_convert(actual) || actual
```

Reply:
> Unless you mean to do the following?

Yep, that's what I had in mind: `actual = Hash.try_convert(actual) || actual`.

Reply:
I made the change as advised, but I'm on the fence as to whether it is an improvement or not. It does look a bit neater (to me), but I don't know if it's as understandable. I mean this in the sense that I have no idea what people would make of either version of the code, what idiom is best known (if known at all) by Ruby devs at large.

I used `#to_hash` directly on my initial version following Avdi Grimm's advice on his book [_Confident Ruby_](http://www.confidentruby.com)[1]. My thinking is that perhaps this is better known and understood because I happen to have read a resource where it was mentioned. I never heard of `.try_convert`, but it might be that it's better known and I just happen not to have read resources on it.

In any case, from Ruby's source code (and quick test), I understand it's the same thing, internally using `#to_hash` where available, so it shouldn't matter functionally whether we use one or the other.

[1] Particularly sections 3.2 _Use built-in conversion protocols_ and 3.3 _Conditionally call conversion methods_.

Reply:
Thanks.  I don't disagree with your reasoning, but weigh other factors.  The main one that makes me think `try_convert` is worth is that it's the official conversion method Ruby has provided.  In the future, it might do more than check `to_hash`.  IMO, it makes sense to use an official conversion method instead of doing it yourself.